diff --git a/Makefile-libbz2_so b/Makefile-libbz2_so
index 9b281eb..bcfeb77 100644
--- a/Makefile-libbz2_so
+++ b/Makefile-libbz2_so
@@ -52,11 +52,13 @@ $(LIBRARY): $(OBJS)
 	$(CC) $(CFLAGS) $(LDFLAGS) -o bzip2-shared $(top_sourcedir)/bzip2.c $(LIBRARY)
 	rm -f libbz2.so.1.0
 	ln -s $(LIBRARY) libbz2.so.1.0
+	ln -s $(LIBRARY) libbz2.so.1
 
 install: all
 	mkdir -p $(DESTDIR)$(root_libdir)
 	install -m755 $(LIBRARY) $(DESTDIR)$(root_libdir)
 	ln -sf $(LIBRARY) $(DESTDIR)$(root_libdir)/libbz2.so.1.0
+	ln -sf $(LIBRARY) $(DESTDIR)$(root_libdir)/libbz2.so.1
 	mkdir -p $(DESTDIR)$(libdir)
 	rm -f $(DESTDIR)$(libdir)/libbz2.so
 	ln -sr $(DESTDIR)$(root_libdir)/$(LIBRARY) $(DESTDIR)$(libdir)/libbz2.so
